#! /bin/sh -e

# DP: Fix a segfault on architectures where the size of a pointer is
# DP: greater than the size of an integer, such as ia64.

dir=
if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
    dir="$3/"
elif [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch)
        patch $pdir -f --no-backup-if-mismatch -p0 < $0
        #cd ${dir}gcc && autoconf
        ;;
    -unpatch)
        patch $pdir -f --no-backup-if-mismatch -R -p0 < $0
        #rm ${dir}gcc/configure
        ;;
    *)
        echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
        exit 1
esac
exit 0

--- libjava/jni/gtk-peer/gnu_java_awt_peer_gtk_GdkGraphics.c.orig	2005-08-02 00:35:30.062697655 -0600
+++ libjava/jni/gtk-peer/gnu_java_awt_peer_gtk_GdkGraphics.c	2005-08-02 00:35:31.721877322 -0600
@@ -46,6 +46,7 @@
 GdkPoint *
 translate_points (JNIEnv *env, jintArray xpoints, jintArray ypoints, 
 		  jint npoints, jint x_offset, jint y_offset);
+GdkPixmap *gnu_java_awt_peer_gtk_GtkImage_getPixmap(JNIEnv *env, jobject obj);
 static void realize_cb (GtkWidget *widget, jobject peer);
 
 JNIEXPORT void JNICALL
